homeassistant/components/xiaomi_miio/light.py

```diff
@@ -774,14 +775,110 @@ def hs_color(self) -> tuple:
+                "Setting brightness and color: "
+                "%s %s%%, %s",
+                brightness, percent_brightness, rgb)
+
+            result = await self._try_command(
+                "Setting brightness and color failed: "
+                "%s bri, %s color",
+                self._light.set_brightness_and_rgb,
+                percent_brightness, rgb)
+
+            if result:
+                self._hs_color = hs_color
+                self._brightness = brightness
+
+        elif ATTR_BRIGHTNESS in kwargs and ATTR_COLOR_TEMP in kwargs:
+            _LOGGER.debug(
+                "Setting brightness and color temperature: "
+                "%s %s%%, %s mireds, %s%% cct",
+                brightness, percent_brightness,
+                color_temp, percent_color_temp)
+
+            result = await self._try_command(
+                "Setting brightness and color temperature failed: "
+                "%s bri, %s cct",
+                self._light.set_brightness_and_color_temperature,
+                percent_brightness, percent_color_temp)
+
+            if result:
+                self._color_temp = color_temp
+                self._brightness = brightness
+
+        elif ATTR_HS_COLOR in kwargs:
+            _LOGGER.debug(
+                "Setting color: %s", rgb)
+
+            result = await self._try_command(
+                "Setting color failed: %s",
+                self._light.set_rgb, rgb)
+
+            if result:
+                self._hs_color = hs_color
+
+        elif ATTR_COLOR_TEMP in kwargs:
+            _LOGGER.debug(
+                "Setting color temperature: "
+                "%s mireds, %s%% cct",
+                color_temp, percent_color_temp)
+
+            result = await self._try_command(
+                "Setting color temperature failed: %s cct",
+                self._light.set_color_temperature, percent_color_temp)
+
+            if result:
+                self._color_temp = color_temp
+
+        elif ATTR_BRIGHTNESS in kwargs:
+            brightness = kwargs[ATTR_BRIGHTNESS]
+            percent_brightness = ceil(100 * brightness / 255.0)
+
+            _LOGGER.debug(
+                "Setting brightness: %s %s%%",
+                brightness, percent_brightness)
+
+            result = await self._try_command(
+                "Setting brightness failed: %s",
+                self._light.set_brightness, percent_brightness)
+
+            if result:
+                self._brightness = brightness
+
+        else:
+            await self._try_command(
+                "Turning the light on failed.", self._light.on)
 
     async def async_update(self):
         """Fetch state from the device."""
         from miio import DeviceException
         try:
             state = await self.hass.async_add_executor_job(self._light.status)
         except DeviceException as ex:
+            if "code" in ex and ex["code"] == -5001:
+                _LOGGER.debug("Device in music mode. Update skipped.")
```

Comment:
I think it would make sense to have this as a one-time warning, wouldn't it? The same error will likely be raised when trying to control the device, too?